pkg/web/http.go

```diff
@@ -0,0 +1,69 @@
+package web
+
+import (
+	"bytes"
+	"crypto/tls"
+	"github.com/gorilla/websocket"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"time"
+)
+
+type WrappedSocketListener struct {
+	sockets chan *WrappedSocket
+	done    chan struct{}
+	addr    net.Addr
+}
+
+func (wsl *WrappedSocketListener) Accept() (net.Conn, error) {
```

Comment:
personal preference, but change `wsl` to a single letter (e.g. `w`); I find multi-letter receiver variable names messy.